wrangler-storage/src/main/java/io/cdap/wrangler/store/upgrade/UpgradeState.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.wrangler.store.upgrade;
+
+/**
+ * Upgrade state to store any upgrade related information. Can be extended to contain more information about the
+ * upgrade information
+ */
+public class UpgradeState {
+  private final boolean isUpgradeComplete;
```

Comment:
It would be better for this to be a version number, so that if we need to upgrade anything in the future we can still use this. 

Similarly, better if the UpgradeStore returns this object and the caller makes the decision of whether or not upgrade needs to run, instead of having the UpgradeStore return a boolean on whether upgrade is done or not.